File: Chapter02/selectSort.js
/* 选择排序 */
/**
 * @description 筛选最小值
 * @param {Array} arr 待筛选数组 
 * @returns {Number}
 */
function findSmallest(arr){
    smallest=arr[0];
    smallestIndex=0;
    for(let i=0;i<arr.length;i++){
        if(smallest>arr[i]){
            smallest=arr[i];
            smallestIndex=i;
        }
    }
    return smallestIndex;
}
/**
 * @description 选择排序
 * @param {Array} arr 
 * @returns {Array}
 */
function selectSort(arr){
    let newArray=[];
    while(arr.length>0){
        smallestIndex=findSmallest(arr);
        let smallestValue=arr.splice(smallestIndex,1)[0];
        newArray.push(smallestValue);
    }
    return newArray;
}
let my_array=[5, 3, 6, 2, 10];
console.log(selectSort(my_array));

File: Chapter02/selectSort.py
# 选择排序
def findSmallest(arr):
    """[查找最小元素]

    Args:
        arr ([list]): [数组]

    Returns:
        [int]: [index]
    """
    smallest=arr[0]
    smallestIndex=0
    for i in range(len(arr)):
        if(smallest>arr[i]):
            smallest=arr[i]
            smallestIndex=i
    return smallestIndex

def selectionSort(arr):
    """[选择排序]

    Args:
        arr ([list]): [待排序数组]

    Returns:
        [type]: [list]
    """
    newArray=[]
    while(len(arr)>0):
        smallestIndex=findSmallest(arr)
        newArray.append(arr.pop(smallestIndex))
    return newArray
my_array=[5, 3, 6, 2, 10]
print(selectionSort(my_array))